src/browser/api_protocol/transport_strategy/elipc_strategy.ts

```diff
@@ -35,15 +35,21 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
             if (strategyName !== this.constructor.name) {
                 next();
             } else if (typeof (action) === 'function') {
-                try {
-                    // singleFrameOnly check first so to prevent frame superceding when disabled.
-                    if (!data.singleFrameOnly === false || e.sender.isValidWithFrameConnect(e.frameRoutingId)) {
-                        action(identity, data, ack, nack);
-                    } else {
-                        nack('API access has been superseded by another frame in this window.');
-                    }
-                } catch (err) {
-                    nack(err);
+                // singleFrameOnly check first so to prevent frame superceding when disabled.
+                if (!data.singleFrameOnly === false || e.sender.isValidWithFrameConnect(e.frameRoutingId)) {
+                    Promise.resolve()
+                        .then(() => action(identity, data, ack, nack))
+                        .then(result => {
+                            if (result !== undefined) {
```

Comment:
in the unlikely case of a null err we still need to nack something, just create a new error and return it, silent fails are never good.